README.md

```diff
@@ -10,16 +23,78 @@ The agent is used through the [StudyBits Wallet](https://github.com/Quintor/Stud
 
 Contact us on [Gitter](https://gitter.im/StudyBits/Lobby)
 
-## Running in docker
 
-Use `TEST_POOL_IP=127.0.0.1 docker-compose up --build --force-recreate pool university-agent-rug university-agent-gent` 
+## Contributing <a name="contributing"></a>
+
+### Repository Structure <a name="structure"></a>
+
+The StudyBits repo consists of the following parts:
+
+```
++ ci
+    + Contains docker files used by trevor for ci.
++ docs
+    + Contains design and implementation documentation.
++ university-agent
+    + Contains java source-code for the university agent.
+```
+
+### Development Setup <a name="dev"></a>
+
+In order to develop for StudyBits a couple of dependencies need to be installed:
+
+1. Install [libindy](https://github.com/hyperledger/indy-sdk/tree/master/libindy)
+    The current version of StudyButs requires Libindy v1.6.6 to be installed, note that later versions are not supperted currently.
+
+2. Make sure [Project Lombok](https://projectlombok.org/) can work it's magic. In your IDE of choice, please enable precompiling.
+
+### How to send a PR <a name="pullrequests"></a>
+
+Before sending a PR make sure to be complient with the following:
 
-## Running tests in docker
++ Do not create many PRs for one feature. Consider implementing a complete feature before sending a PR. 
++ Consider sending a design doc folder for a new feature.
++ Make sure that a new feature or fix is covered by tests (try following TDD).
++ Make sure that documentation is updated according to your changes.
++ Provide a full description of changes in the PR.
++ Make sure that static code validation passed.
++ You need to make sure that all the tests pass.
++ A reviewer or maintainer will merge the PR.
+
+## Running StudyBits <a name="running"></a>
+
+### Running in docker <a name="RunInDocker"></a>
+
+Use `TEST_POOL_IP=127.0.0.1 docker-compose up --build --force-recreate pool university-agent-rug university-agent-gent` 
 
 Running tests: `TEST_POOL_IP=127.0.0.1 docker-compose up --build --force-recreate --exit-code-from tests`
 
-## Running outside of docker
+Please note the the enviroment variable TEST_POOL_IP must be supplied as the IP of the HyperLedger indy nodes you wish to use.
+
+## System Diagrams <a name="diagrams"></a>
+
+![PackageModel](/docs/images/PackageDiagramAgent.png "UniversityAgent Package Diagram")
```

Comment:
I think this is more suited for a separate page, perhaps a .md in the docs folder that we can link here. 

For all other pictures, I'd either remove them, or reference them in a markdown file, otherwise they're just invisible.